project-base/tests/FrontendApiBundle/Functional/Order/GetOrdersAsAuthenticatedCustomerUserTest.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrontendApiBundle\Functional\Order;
+
+use Tests\FrontendApiBundle\Test\GraphQlWithLoginTestCase;
+
+class GetOrdersAsAuthenticatedCustomerUserTest extends GraphQlWithLoginTestCase
+{
+    /**
+     * @param string $query
+     * @param array $expectedOrdersData
+     * @dataProvider getOrdersDataProvider
+     */
+    public function testGetAllCustomerUserOrders(string $query, array $expectedOrdersData): void
+    {
+        $graphQlType = 'orders';
+        $response = $this->getResponseContentForQuery($query);
+        $this->assertResponseContainsArrayOfDataForGraphQlType($response, $graphQlType);
+        $responseData = $this->getResponseDataForGraphQlType($response, $graphQlType);
+
+        $this->assertArrayHasKey('edges', $responseData);
+        $this->assertCount(count($expectedOrdersData), $responseData['edges']);
+
+        foreach ($responseData['edges'] as $edge) {
+            $this->assertArrayHasKey('node', $edge);
+
+            $expectedOrderData = array_shift($expectedOrdersData);
+            $this->assertArrayHasKey('status', $edge['node']);
+            $this->assertSame($expectedOrderData['status'], $edge['node']['status']);
+
+            $this->assertArrayHasKey('totalPrice', $edge['node']);
+            $this->assertArrayHasKey('priceWithVat', $edge['node']['totalPrice']);
+            $this->assertSame($expectedOrderData['priceWithVat'], $edge['node']['totalPrice']['priceWithVat']);
+        }
+    }
+
+    /**
+     * @return array
+     */
+    public function getOrdersDataProvider(): array
+    {
+        return [
+            [
+                $this->getAllOrdersQuery(),
+                $this->getExpectedUserOrders(),
+            ],
+            [
+                $this->getFirstOrdersQuery(2),
+                array_slice($this->getExpectedUserOrders(), 0, 2),
+            ],
+            [
+                $this->getFirstOrdersQuery(1),
+                array_slice($this->getExpectedUserOrders(), 0, 1),
+            ],
+            [
+                $this->getLastOrdersQuery(1),
+                array_slice($this->getExpectedUserOrders(), 5, 1),
+            ],
+            [
+                $this->getLastOrdersQuery(2),
+                array_slice($this->getExpectedUserOrders(), 4, 2),
+            ],
+        ];
+    }
+
+    /**
+     * @return string
+     */
+    private function getAllOrdersQuery(): string
```

Comment:
This is actually getFirstTenOrdersQuery :)

Reply:
Yeah, that is really tricky 👍